server/Dockerfile

```
@@ -0,0 +1,44 @@
+RUN addgroup -S spring && adduser -S spring -G spring
+
+# Copy JAR from builder
+COPY --from=builder /app/target/*.jar app.jar
+
+# Change ownership
+RUN chown -R spring:spring /app
+
+USER spring
+
+# Port (Render uses PORT env var, default 4000 matches application.properties)
+EXPOSE ${PORT:-4000}
```

Hadolint (2.15.1)

[error] unexpected '}'
expecting '/', a new line followed by the next instruction, or digit

(DL1000)